morebits.js

```diff
@@ -2041,6 +2044,21 @@ Morebits.wiki.page = function(pageName, currentAction) {
 		ctx.followRedirect = followRedirect;
 	};
 
+	// lookup-creation setter function
+	/**
+	 * @param {boolean} flag - if set true, the author and timestamp of the first non-redirect
+	 * version of the page is retrieved.
+	 *
+	 * Warning:
+	 * 1. If there are no revisions among the first 50 that are non-redirects, or if there are
+	 *    less 50 revisions and all are redirects, the original creation is retrived.
+	 * 2. Revisions that the user is not privileged to access (revdeled/suppressed) will be treated
+	 *    as non-redirects.
+	 */
```

Comment:
This makes sense as revdels in main namespace is usually for copyright violations, and only non-redirect revisions would have been revdeled as such.

Reply:
Doesn't really make a difference either way: not much else we can do, and it should be pretty rare regardless.  The reality is that early vandalism is going to be an issue no matter what, as articles that were initially redirects but had someone vandalize that redirect before becoming an article will still notify the wrong person.  In the end, though, this is basically taking a simple process and making it somewhat smarter, but that doesn't mean we need to account for every single contingency.